tests/cli/deployment/test_deployment_run.py

```diff
@@ -0,0 +1,212 @@
+        ("27 hours + 4 mins", pendulum.duration(days=1, hours=3, minutes=4)),
+    ],
+)
+async def test_start_in_option_schedules_flow_run_in_future(
+    deployment_name: str,
+    frozen_now,
+    orion_client: prefect.OrionClient,
+    start_in: str,
+    expected_duration,
+):
+    expected_start_time = frozen_now + expected_duration
+    expected_display = expected_start_time.in_tz(
+        pendulum.tz.local_timezone()
+    ).to_datetime_string()
+
+    await run_sync_in_worker_thread(
+        invoke_and_assert,
+        command=[
+            "deployment",
+            "run",
+            deployment_name,
+            "--start-in",
+            start_in,
+        ],
+        expected_output_contains=f"Scheduled start time: {expected_display}",
+    )
+
+    flow_runs = await orion_client.read_flow_runs()
+    assert len(flow_runs) == 1
+    flow_run = flow_runs[0]
+
+    assert flow_run.state.is_scheduled()
+    scheduled_time = flow_run.state.state_details.scheduled_time
+
+    assert scheduled_time == expected_start_time
+
+
+@pytest.mark.parametrize(
+    "start_in, expected_display",
+    [
+        ("10 minutes", "in 10 minutes"),
+        ("5 days", "in 5 days"),
+        ("3 seconds", "in a few seconds"),
+        (None, "now"),
+        ("1 year and 3 months", "in 1 year"),
+        ("2 weeks & 1 day", "in 2 weeks"),
+        ("27 hours + 4 mins", "in 1 day"),
+    ],
+)
+async def test_start_in_displays_scheduled_start_time(
+    deployment_name: str,
+    orion_client: prefect.OrionClient,
+    start_in: str,
+    expected_display: str,
+):
+
+    await run_sync_in_worker_thread(
+        invoke_and_assert,
+        command=[
+            "deployment",
+            "run",
+            deployment_name,
+            "--start-in",
+            start_in,
+        ],
+        expected_output_contains=["Scheduled start time:", expected_display],
+    )
+
+    flow_runs = await orion_client.read_flow_runs()
+    assert len(flow_runs) == 1
+    flow_run = flow_runs[0]
+
+    assert flow_run.state.is_scheduled()
+
+
+def test_start_at_displays_scheduled_start_time(deployment_name: str, monkeypatch):
+    monkeypatch.setattr(
+        "pendulum.now",
+        lambda *_: pendulum.instance(dateparser.parse("January 1st 2023")),
+    )
+    invoke_and_assert(
+        command=[
+            "deployment",
+            "run",
+            deployment_name,
+            "--start-at",
+            "January 1st 2022",
+        ],
+        expected_output_contains="Scheduled start time: 2022-01-01 00:00:00 (1 year ago)",
+    )
+
+
+def test_start_at_invalid_input(deployment_name: str):
+    invoke_and_assert(
+        command=[
+            "deployment",
+            "run",
+            deployment_name,
+            "--start-at",
+            "foobar",
+        ],
+        expected_code=1,
+        expected_output="Unable to parse scheduled start time 'at foobar'.",
+    )
+
+
+def test_start_in_invalid_input(deployment_name: str):
```

Comment:
We may want to try covering some more invalid input types here, like malformed timestamps.